File: leetcode_58/solution.py
class Solution:
    def lengthOfLastWord(self, s):
        """
        :type s: str
        :rtype: int
        """
        # s, _len = s.strip(), 0
        # for i in range(len(s) - 1, -1, -1):
        #     if s[i] == ' ':
        #         break
        #     _len += 1
        # return _len

        # return len(s.strip().split(' ').pop())

        j = len(s) - 1
        while j > -1 and s[j] == ' ':
            j -= 1

        l = 0
        while j > -1 and s[j] != ' ':
            l += 1
            j -= 1

        return l

File: leetcode_58/solution.go
package main

import "fmt"

//func lengthOfLastWord(s string) int {
//	if s == "" {
//		return 0
//	}
//
//	j := len(s) - 1
//	for j > -1 {
//		if s[j] != ' ' {
//			break
//		} else {
//			j--
//		}
//	}
//
//	i := j
//	for i > -1 {
//		if s[i] == ' ' {
//			break
//		} else {
//			i--
//		}
//	}
//
//	return j - i
//}

func lengthOfLastWord(s string) int {
	if s == "" {
		return 0
	}

	j := len(s) - 1
	for j > -1 && s[j] == ' ' {
		j--
	}

	l := 0
	for j > -1 && s[j] != ' ' {
		l++
		j--
	}

	return l
}

func main() {
	s := "  a "
	fmt.Println(lengthOfLastWord(s))
}